modules/apps/27-interchain-accounts/controller/keeper/events.go

```diff
@@ -13,22 +15,18 @@ import (
 
 // EmitAcknowledgementEvent emits an event signalling a successful or failed acknowledgement and including the error
 // details if any.
-func EmitAcknowledgementEvent(ctx context.Context, packet channeltypes.Packet, ack exported.Acknowledgement, err error) {
-	sdkCtx := sdk.UnwrapSDKContext(ctx) // TODO: https://github.com/cosmos/ibc-go/issues/5917
-	attributes := []sdk.Attribute{
-		sdk.NewAttribute(sdk.AttributeKeyModule, icatypes.ModuleName),
-		sdk.NewAttribute(icatypes.AttributeKeyControllerChannelID, packet.GetDestChannel()),
-		sdk.NewAttribute(icatypes.AttributeKeyAckSuccess, strconv.FormatBool(ack.Success())),
+func (k *Keeper) EmitAcknowledgementEvent(ctx context.Context, packet channeltypes.Packet, ack exported.Acknowledgement, err error) {
+	attributes := []event.Attribute{
+		event.NewAttribute(sdk.AttributeKeyModule, icatypes.ModuleName),
+		event.NewAttribute(icatypes.AttributeKeyControllerChannelID, packet.GetDestChannel()),
+		event.NewAttribute(icatypes.AttributeKeyAckSuccess, strconv.FormatBool(ack.Success())),
 	}
 
 	if err != nil {
-		attributes = append(attributes, sdk.NewAttribute(icatypes.AttributeKeyAckError, err.Error()))
+		attributes = append(attributes, event.NewAttribute(icatypes.AttributeKeyAckError, err.Error()))
 	}
 
-	sdkCtx.EventManager().EmitEvent(
-		sdk.NewEvent(
-			icatypes.EventTypePacket,
-			attributes...,
-		),
-	)
+	if err := k.EventService.EventManager(ctx).EmitKV(icatypes.EventTypePacket, attributes...); err != nil {
+		panic(err)
+	}
```

Comment:
This error should be propagated and returned from this func. 

The likelihood of these errors surfacing from `EventManager` are pretty much zero, but at some point in a future far away it may be possible for it to error - given a different impl of event service.

None the less, its good practise to actually return the errors and not panic and rely on grpc recovery interceptor

Reply:
Should it have a todo comment for that, then?

Reply:
https://github.com/cosmos/ibc-go/issues/7624